game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProOddsCalculator.java

```diff
@@ -217,23 +228,40 @@ public ProBattleResult callBattleCalculator(
     double tuvSwing =
         results.getAverageTuvSwing(
             attacker, mainCombatAttackers, defender, mainCombatDefenders, data);
-    if (Matches.territoryIsNeutralButNotWater().test(t)) { // Set TUV swing for neutrals
+
+    // Set TUV swing for neutrals
+    if (Matches.territoryIsNeutralButNotWater().test(t)) {
       final double attackingUnitValue = TuvUtils.getTuv(mainCombatAttackers, ProData.unitValueMap);
       final double remainingUnitValue =
           results
               .getAverageTuvOfUnitsLeftOver(ProData.unitValueMap, ProData.unitValueMap)
               .getFirst();
       tuvSwing = remainingUnitValue - attackingUnitValue;
     }
+
+    // Add TUV swing for transported units
     final List<Unit> defendingTransportedUnits =
         CollectionUtils.getMatches(defendingUnits, Matches.unitIsBeingTransported());
-    if (t.isWater()
-        && !defendingTransportedUnits.isEmpty()) { // Add TUV swing for transported units
+    if (t.isWater() && !defendingTransportedUnits.isEmpty()) {
       final double transportedUnitValue =
           TuvUtils.getTuv(defendingTransportedUnits, ProData.unitValueMap);
       tuvSwing += transportedUnitValue * winPercentage / 100;
     }
 
+    // Remove TUV and add to remaining units for defenders that can submerge before battle
+    if (tuvSwing > 0
+        && checkSubmerge
```

Comment:
are all 4 of these boolean conditions to represent "canSubmergeBeforeBattle"? If so, extracting to a variable or method could make the comment redundant and the code a bit easier to grok

Reply:
Agree. Updated.